From: Bruno Randolf <br1@einfach.org>
Date: Thu, 25 Mar 2010 14:49:31 +0900
Subject: ath5k: initialize default noise floor

commit 9d332c82b4cf2e4538450e4af40f073cc5e599ec upstream.

Initialize noise floor variable with a default of -95. This was used
uninitialized in the signal strength (RSSI -> dBm) conversion until the first
noise floor calibration was completed.

Signed-off-by: Bruno Randolf <br1@einfach.org>
Signed-off-by: John W. Linville <linville@tuxdriver.com>
Signed-off-by: Jonathan Nieder <jrnieder@gmail.com>
---
 drivers/net/wireless/ath/ath5k/attach.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath5k/attach.c b/drivers/net/wireless/ath/ath5k/attach.c
index 88663dfe3b3d..b5ea906cd7c8 100644
--- a/drivers/net/wireless/ath/ath5k/attach.c
+++ b/drivers/net/wireless/ath/ath5k/attach.c
@@ -134,6 +134,7 @@ struct ath5k_hw *ath5k_hw_attach(struct ath5k_softc *sc)
 	ah->ah_limit_tx_retries = AR5K_INIT_TX_RETRY;
 	ah->ah_software_retry = false;
 	ah->ah_current_channel = &sc->channels[0];
+	ah->ah_noise_floor = -95;	/* until first NF calibration is run */
 
 	/*
 	 * Find the mac version
-- 
1.7.10

